packages/react/src/components/Forms/FileUploader/index.spec.tsx

```diff
@@ -0,0 +1,557 @@
+import { act, render, screen } from "@testing-library/react";
+import React, { useRef, useState } from "react";
+import userEvent from "@testing-library/user-event";
+import { CunninghamProvider } from ":/components/Provider";
+import {
+  FileUploader,
+  FileUploaderRefType,
+} from ":/components/Forms/FileUploader/index";
+import { Button } from ":/components/Button";
+
+/**
+ * There are limitations in this test:
+ * jestdom does not support replacing input.files with another FileList, which is what we do in the component when
+ * we delete a file from the list for example.
+ * So what we do is we mock the utils module and replace the replaceInputFilters function with a no-op. The consequence
+ * of this is that the real files ( as value ) of the input are not replaced.
+ */
+vi.mock("./utils", async () => {
+  const actual: any = await vi.importActual("./utils");
+  return { ...actual, replaceInputFilters: () => {} };
+});
+
+describe("<FileUploader />", () => {
+  describe("Mono", () => {
+    it("should select a file and display its name", async () => {
+      render(
+        <CunninghamProvider>
+          <FileUploader />
+        </CunninghamProvider>
+      );
+      const user = userEvent.setup();
+      const input: HTMLInputElement =
+        screen.getByLabelText(/Drag and drop or /);
```

Comment:
It looks weird for a11y purpose

Reply:
As far as I got informations we can't select a file input with just a pure "role" 😅